07-routing-gossip.md

```diff
@@ -15,12 +15,34 @@ information), and at least two `channel_update` messages are expected.
 
 ## The `channel_announcement` message
 
-This message contains ownership information about a channel.  It ties each
-on-chain Bitcoin key to the lightning node key, and vice-versa.
-
+This message contains ownership information about a channel.  It ties
+each on-chain Bitcoin key to the lightning node key, and vice-versa.
 The channel is not really usable until at least one side has announced
 its fee levels and expiry using `channel_update`.
 
+In order to prove the existence of channel between `node-1` and
+`node-2` we need to:
+
+1. Prove that the funding transaction pays to `bitcoin-key-1` and
+   `bitcoin-key-2`.
+2. Prove `node-1` owns `bitcoin-key-1`
+3. Prove `node-2` owns `bitcoin-key-2`
+
+The first one is done by assuming that all nodes know the unspent
+transaction outputs, and thus can find the output given by
+`channel-id` and validate that it is indeed a P2WSH funding
+transaction output as to those keys specified in
+[BOLT #3](03-transactions.md#funding-transaction-output).
+
+The second two are done by explicit signatures (`bitcoin-signature-1`
+and `bitcoin-signature-2`, generated by each `bitcoin-key` and signing
+the corresponding `node-id`.
+
+We also need to prove that `node-1` and `node-2` both agree on this
+announcement message; that is done by having a signature from each
+`node-id` signing the entire message (`node-signature-1` and
+`node-signature-2`).
```

Comment:
This is a bit underdefined, we can either set the signatures to a fixed value (all zeros) and then sign the entire message, or we can sign the message starting from byte 128, i.e., skipping the signature fields themselves. I have a slight preference for the latter since then we can just store the signatures in-place and skip serializing twice. In the former case we'd have to store the signatures somewhere else while doing the two signatures and then copy them over.